.github/workflows/build-macos.yaml

```diff
@@ -0,0 +1,26 @@
+name: Build on Mac OS
+run-name: Build on Mac OS
+
+on:
+  pull_request:
+  push:
+  
+jobs:
+  build:
+    name: Build tsh on Mac OS
+    runs-on: macos-12
+
+    permissions:
+      contents: read
+
+    steps:
+      - name: Checkout Teleport
+        uses: actions/checkout@v3
+
+      - name: Setup Go
```

Comment:
Should we install rust too? If we don't, desktop access won't be built

Reply:
MacOS 12 and windows-latest images have the latest Rust installed by default: https://github.com/actions/runner-images/blob/main/images/macos/macos-12-Readme.md Default go version is 1.17, though. That's the reason why we set it up separately.